### PR TITLE
[tt-train] Add reduction type None to CrossEntropyLoss

### PR DESCRIPTION
**Pull request by rfurko-tt**
### Problem description
This will help our fine-tune demo. 

### What's changed
* Add reduction type None to CrossEntropyLoss
* Add test for None vs Mean reduction type 

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/18389368067) CI passes
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by dmakoviichuk-tt — APPROVED** [review]

File: tt-train/sources/ttml/ops/losses.hpp [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 namespace ttml::ops {
 
-enum ReduceType : uint8_t { MEAN = 0, SUM = 1 };
+enum ReduceType : uint8_t { NONE = 0, MEAN = 1, SUM = 2 };
```
Comment (abustamanteTT) [inline tt-train/sources/ttml/ops/losses.hpp:11]: Could you update the loss nanobind so that it supports None as well?

Comment (rfurko-tt) [reply tt-train/sources/ttml/ops/losses.hpp:11]:
```
        py_loss.def(
            "cross_entropy_loss",
            &ttml::ops::cross_entropy_loss,
            nb::arg("prediction"),
            nb::arg("target"),
            nb::arg("reduce") = ReduceType::MEAN);
```

we already pass ReduceType as a parameter to cross_entropy_loss in python. should work already

Comment (abustamanteTT) [reply tt-train/sources/ttml/ops/losses.hpp:11]: Thank you, I missed the fact that ReduceType is already bound and any updates to the enum will be built in

**Review by abustamanteTT — APPROVED** [review]
